server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -584,6 +584,7 @@ private void sendBatch(
             final long maxSeqNoOfUpdatesOrDeletes,
             final RetentionLeases retentionLeases,
             final ActionListener<Long> listener) throws IOException {
+        ThreadPool.assertCurrentMethodIsNotCalledRecursively();
```

Comment:
this should be ` assert ThreadPool.assertCurrentMethodIsNotCalledRecursively();`

Reply:
Yes, fixed in https://github.com/elastic/elasticsearch/pull/39988/commits/6109f0ee68ac8fe7760bfcf8206d95964cbf9c87.